test/parallel/test-watch-mode-files_watcher.mjs

```diff
@@ -95,67 +92,60 @@ describe('watch mode file watcher', () => {
     assert.strictEqual(changesCount, 1);
   });
 
-  it('should watch all files in watched path when in "all" mode',
-     { skip: !supportsRecursiveWatching }, async () => {
-       watcher = new FilesWatcher({ throttle: 100, mode: 'all' });
-       watcher.on('changed', () => changesCount++);
-
-       const file = path.join(tmpdir.path, 'file5');
-       watcher.watchPath(tmpdir.path);
-
-       const changed = once(watcher, 'changed');
-       writeFileSync(file, 'changed');
-       await changed;
-       assert.strictEqual(changesCount, 1);
-     });
-
-  it('should ruse existing watcher if it exists',
-     { skip: !supportsRecursiveWatching }, () => {
-       assert.deepStrictEqual(watcher.watchedPaths, []);
-       watcher.watchPath(tmpdir.path);
-       assert.deepStrictEqual(watcher.watchedPaths, [tmpdir.path]);
-       watcher.watchPath(tmpdir.path);
-       assert.deepStrictEqual(watcher.watchedPaths, [tmpdir.path]);
-     });
-
-  it('should ruse existing watcher of a parent directory',
-     { skip: !supportsRecursiveWatching }, () => {
-       assert.deepStrictEqual(watcher.watchedPaths, []);
-       watcher.watchPath(tmpdir.path);
-       assert.deepStrictEqual(watcher.watchedPaths, [tmpdir.path]);
-       watcher.watchPath(path.join(tmpdir.path, 'subdirectory'));
-       assert.deepStrictEqual(watcher.watchedPaths, [tmpdir.path]);
-     });
-
-  it('should remove existing watcher if adding a parent directory watcher',
-     { skip: !supportsRecursiveWatching }, () => {
-       assert.deepStrictEqual(watcher.watchedPaths, []);
-       const subdirectory = path.join(tmpdir.path, 'subdirectory');
-       mkdirSync(subdirectory);
-       watcher.watchPath(subdirectory);
-       assert.deepStrictEqual(watcher.watchedPaths, [subdirectory]);
-       watcher.watchPath(tmpdir.path);
-       assert.deepStrictEqual(watcher.watchedPaths, [tmpdir.path]);
-     });
-
-  it('should clear all watchers when calling clear',
-     { skip: !supportsRecursiveWatching }, () => {
-       assert.deepStrictEqual(watcher.watchedPaths, []);
-       watcher.watchPath(tmpdir.path);
-       assert.deepStrictEqual(watcher.watchedPaths, [tmpdir.path]);
-       watcher.clear();
-       assert.deepStrictEqual(watcher.watchedPaths, []);
-     });
+  it('should watch all files in watched path when in "all" mode', async () => {
+    watcher = new FilesWatcher({ throttle: 100, mode: 'all' });
+    watcher.on('changed', () => changesCount++);
+
+    const file = path.join(tmpdir.path, 'file5');
+    watcher.watchPath(tmpdir.path);
+
+    const changed = once(watcher, 'changed');
+    writeFileSync(file, 'changed');
```

Comment:
Can you add a timeout before writeFileSync in here? `watch` is an async operation